ale/kernel_access.py

```diff
@@ -295,3 +298,65 @@ def remove_dups(listofElements):
     else:
         warnings.warn(f"{format_as} is not a valid format, returning as dict")
         return kernels
+
+
+def spiceql_call(function_name = "", function_args = {}, use_web=False):
+    """
+    Interface to SpiceQL (Spice Query Library) for both Offline and Online use
+
+    This function will access the value passed through props defined as `web`. This
+    value determines the access pattern for spice data. When set to Online, you will
+    access the SpiceQL service provided through the USGS Astro AWS platform. This service
+    performs kernel and data aquisition. If set to Offline, you will access locally loaded
+    kernels, and SpiceQL will do no searching for you.
+
+    Parameters
+    ----------
+    functions_name : str
+                        String defineing the function to call, properly exposed SpiceQL
+                        functions should map 1-to-1 with endpoints on the service
+                    
+    function_args : dict
+                    Dictionary of arguments used by the function
+
+    Returns : any
+                Any return from a SpiceQL function
+    """
+    if not use_web:
+        func = getattr(pyspiceql, function_name)
+
+        # Ensure that in offline mode we anticipate the user loading/passing their own kernels
+        # to ALE
+        return func(**function_args)
+
+    url = "https://spiceql-dev.prod-asc.chs.usgs.gov/v1/"
+    url += function_name
+    headers = {
+        'accept': '*/*',
+        'Content-Type': 'application/json'
+    }
+
+    r = requests.get(url, params=function_args, headers=headers, verify=False)
```

Comment:
Yes, this verify=False is needed on a DOI network if the SSL certificates are not in place, but out in the wild this is a bad idea.

Reply:
Both good catches!

Reply:
Revisiting this. Isn't `requests` is a default python library?

Reply:
No, it is not. If you want to use just standard library, use urllib (though it does not have the same nice to have stuff like urllib3 or requests).